src/main/java/appeng/recipes/AERecipeLoader.java

```diff
@@ -0,0 +1,121 @@
+
+package appeng.recipes;
+
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.HashMap;
+import java.util.Map;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParseException;
+import com.google.gson.JsonSyntaxException;
+
+import org.apache.commons.io.FilenameUtils;
+import org.apache.commons.io.IOUtils;
+
+import net.minecraft.util.JsonUtils;
+import net.minecraft.util.ResourceLocation;
+import net.minecraftforge.common.crafting.CraftingHelper;
+import net.minecraftforge.common.crafting.JsonContext;
+import net.minecraftforge.fml.common.FMLLog;
+import net.minecraftforge.fml.common.Loader;
+import net.minecraftforge.fml.common.ModContainer;
+
+import appeng.core.AppEng;
+import appeng.recipes.handlers.GrinderHandler;
+import appeng.recipes.handlers.InscriberHandler;
+import appeng.recipes.handlers.SmeltingHandler;
+
+
+public class AERecipeLoader
+{
+	private static final String AERECIPE_BASE = "/aerecipes";
+	private static Gson GSON = new GsonBuilder().setPrettyPrinting().disableHtmlEscaping().create();
+	private final Map<ResourceLocation, IAERecipeFactory> factories = new HashMap<>();
+
+	private final ModContainer mod;
+	private final JsonContext ctx;
+
+	public AERecipeLoader()
+	{
+		mod = Loader.instance().getIndexedModList().get( AppEng.MOD_ID );
+		ctx = new JsonContext( AppEng.MOD_ID );
+
+		initFactories();
+	}
+
+	public boolean loadProcessingRecipes()
+	{
+		return CraftingHelper.findFiles( this.mod, "assets/" + AppEng.MOD_ID + AERECIPE_BASE, this::preprocess, this::process, true, true );
+	}
+
+	private boolean preprocess( final Path root )
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this unused method parameter "root". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1172)